File: Assets/Extensions/FrispAds/AdUnits/AppleAd.cs

using UnityEngine;
using System.Collections;

namespace FrispAds.AdUnits {

	using ADBannerView = FrispAds.Api.ADBannerView;

	public class AppleAd : AdUnit {
		
		private ADBannerView banner;
		private bool showAd = true;
		private bool loaded = false;
		
		public AppleAd() {
			banner = ADBannerView.instance ();
			banner.CreateBanner ();
			banner.BannerViewDidLoadAd += BannerViewDidLoadAd;
			banner.DidFailToReceiveAdWithError += DidFailToReceiveAdWithError;
		}

		public void Hide() {
			showAd = false;
			banner.Hide ();
		}
		
		public void Show() {
			showAd = true;
			banner.Show ();
		}
		
		public bool Loaded() {
			return banner.Loaded ();
		}

		public void BannerViewDidLoadAd() {
			loaded = true;
			if (showAd) {
				Show ();
			}
		}

		public void DidFailToReceiveAdWithError() {
			loaded = false;
		}
	}
}

File: Assets/Plugins/iOS/FrispGames/Ads/AppleAd.m
#import "AppleAd.h"

@implementation AppleAd

- (void) SetupBanner {
  [self setBanner: [[ADBannerView alloc] initWithFrame:CGRectZero]];
  [[self banner] setAutoresizingMask:UIViewAutoresizingFlexibleWidth];
  [self banner].delegate = self;
  [[self banner] setBackgroundColor:[UIColor clearColor]];
}

- (void) CreateBanner {
  [self SetupBanner];
  
  UIViewController *rootViewController =  [[[[UIApplication sharedApplication]delegate] window] rootViewController];
  
  NSSet *contentSizeSet = [NSSet setWithObjects:ADBannerContentSizeIdentifierLandscape,ADBannerContentSizeIdentifierPortrait, nil];

  [[self banner] setRequiredContentSizeIdentifiers: contentSizeSet];

  [self resetContentSize: rootViewController.interfaceOrientation];
  
  [self banner].frame = CGRectMake(0,
                                   rootViewController.view.frame.size.height - _banner.frame.size.height,
                                   [self banner].frame.size.width,
                                   [self banner].frame.size.height);
  
  [[rootViewController view] addSubview: [self banner]];
  
  [self banner].hidden = true;
}

- (void) HideAd {
  [self banner].hidden = true;
}

- (void) ShowAd {
  [self banner].hidden = false;
}

- (BOOL) AdLoaded {
  return [self banner].bannerLoaded;
}

-(void)bannerView:(ADBannerView *)banner didFailToReceiveAdWithError:(NSError *)error {
  if(error != nil) {
       NSLog(@"didFailToReceiveAdWithError %@", error.description);
  }
  
  UnitySendMessage("ADBannerView", "didFailToReceiveAdWithError", "");
}

- (void) bannerViewDidLoadAd:(ADBannerView *)banner {
  NSLog(@"bannerViewDidLoadAd");
  UnitySendMessage("ADBannerView", "bannerViewDidLoadAd", "");
}

- (void) bannerViewWillLoadAd:(ADBannerView *)banner {
  NSLog(@"bannerViewWillLoadAd");  
  UnitySendMessage("ADBannerView", "bannerViewWillLoadAd", "");
}

- (void) bannerViewActionDidFinish:(ADBannerView *)banner {
  NSLog(@"bannerViewActionDidFinish");
  UnitySendMessage("ADBannerView", "bannerViewActionDidFinish", "");
}

- (BOOL) bannerViewActionShouldBegin:(ADBannerView *)banner willLeaveApplication:(BOOL)willLeave {
  NSLog(@"bannerViewActionShouldBegin");
  UnitySendMessage("ADBannerView", "bannerViewActionShouldBegin", "");
  return true;
}

@end
